File: oldSnacks/CreditCardValidator.java
import java.util.Scanner;

public class CreditCardValidator {

   public static String getCardType(String number) {

      if (number.startsWith("4") ) {
         return "Visa";
           }
      else if (number.startsWith("5")) {
          return "MasterCard";
             }
      else if (number.startsWith("37")) {
          return "American Express";
             }
      else if (number.startsWith("6")) {
          return "Discover";
               }
      else {
          return "Invalid Card Type";
               }

              }

    public static boolean isLuhnValid(String number) {
         int sum = 0;
         boolean isEven = false;

         for (int i = number.length() - 1; i >= 0; i--) {
            char digitChar = number.charAt(i);
            if (Character.isDigit(digitChar)) {
              int digit = Character.getNumericValue(digitChar);
              if (isEven) {
                digit *= 2;
               if (digit >= 10) {
                 digit = digit % 10 + digit / 10;
                }
               }
               sum += digit;
               isEven = !isEven;
             }
            }
              
             return sum % 10 == 0;
            }

    public static String validateCard(String number) {
        String cardType = getCardType(number);
        if (cardType.equals("Invalid Card Type")) {
           return "Invalid Card Type: " + number;
           }

        if (isLuhnValid(number)) {
          return cardType + " card - Valid: " + number;
            }
        else {
          return cardType + " card - Invalid: " + number;
            }
          }

    public static void main(String [] args) {
      Scanner scanner = new Scanner(System.in);
      System.out.print("Enter your credit card number: ");
      String cardNumber = scanner.nextLine();

      String validationResult = validateCard(cardNumber);
      System.out.println(validationResult);
       }
     }

File: oldSnacks/credit_card_validator.py
card_number = input("Enter your credit card number: ")

if card_number.startswith("4"):
	card_type = "Visa"
elif card_number.startswith("5"):
	card_type = "MasterCard"
elif card_number.startswith("37"):
	card_type = "American Express"
elif card_number.startswith("6"):
	card_type = "Discover"
else:
	card_type = "Invalid Card Type"

sum = 0
is_even = False
for i in range(len (card_number) -1, -1, -1):
	char = card_number[i]
	if char.isdigit():
		digit = int(char)
		if is_even:
			digit *= 2
			digit = digit % 10 + digit // 10 if digit >= 10 else digit
		sum += digit
		is_even = not is_even


if card_type == "Invalid Card Type":
	print(f"Invalid Card Type: {card_number}")
elif sum % 10 == 0:
	print(f"{card_type} card - Valid: {card_number}")
else:
	print(f"{card_type} card_Invalid: {card_number}")